src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/Helpers.cs

```diff
@@ -106,14 +113,30 @@ internal static bool IsFilterCharacter(RoslynCompletionItem item, char ch, strin
             }
 
             // general rule: if the filtering text exactly matches the start of the item then it must be a filter character
-            if (CommitManager.TextTypedSoFarMatchesItem(item, textTypedSoFar))
+            if (TextTypedSoFarMatchesItem(item, textTypedSoFar))
             {
                 return true;
             }
 
             return false;
         }
 
+        internal static bool TextTypedSoFarMatchesItem(RoslynCompletionItem item, string textTypedSoFar)
+        {
+            if (textTypedSoFar.Length > 0)
+            {
+                // Note that StartsWith ignores \0 at the end of textTypedSoFar on VS Mac and Mono.
+                return item.DisplayText.StartsWith(textTypedSoFar, StringComparison.CurrentCultureIgnoreCase) ||
+                       item.FilterText.StartsWith(textTypedSoFar, StringComparison.CurrentCultureIgnoreCase);
+            }
+
+            return false;
+        }
+
+        // Tab, Enter and Null (call invoke commit) are always commit characters. 
+        internal static bool IsStandardCommitCharacter(char c)
+            => c == '\t' || c == '\n' || c == '\0';
```

Comment:
it is super frustrating that 'null' is used for this purpose.  This exact case was brought up originally with the new completion system and it was pointed out that (ab)(mis)using commit characters in this way would be error prone.  The right way to handle this is to simply have the actual completion system demarcate out a set of commit triggers (like the roslyn system did) that way we're not just having to interpret a space of 65k values to figure out which are relevant in all teh different contexts.  Having special sentinel character that aren't actually characters is not beneficial and only leads to bugs (as we see here).

in terms of the fix, this is fine in that it's trivial and can be done now.  but we shouldn't abuse domains like this (esp. as we had a perfectly fine design before that wasn't subject to this sort of issue).

Reply:
Thank you for the feedback, @CyrusNajmabadi ! Adding @AmadeusW to review. Maybe we can make more strongly typed.